src/Language/PureScript/Sugar/Names.hs

```diff
@@ -350,11 +350,8 @@ renameInModule imports (Module ss coms mn decls exps) =
       -- in scope, we throw an error.
       (Just options, _) -> do
         (mnNew, mnOrig) <- checkImportConflicts mn toName options
-        modify $ \result ->
-          M.insert
-            mnNew
-            (maybe [fmap toName qname] (fmap toName qname :) (mnNew `M.lookup` result))
-            result
+        modify $ \usedImports ->
+          M.insertWith (++) mnNew [fmap toName qname] usedImports
```

Comment:
I see where you're coming from, but I think this is a relatively straightforward use of point-free style.